pkgs/build-support/node/fetch-pnpm-deps/default.nix

```diff
@@ -0,0 +1,89 @@
+{
+  stdenvNoCC,
+  fetchurl,
+  jq,
+  moreutils,
+  cacert,
+  makeSetupHook,
+}:
+{
+  mkFetchPnpmDeps = pnpm: {
+    fetchDeps =
+      {
+        src,
+        hash ? "",
+        pname,
+        ...
+      }@args:
+      let
+        args' = builtins.removeAttrs args [
+          "hash"
+          "pname"
+        ];
+        hash' =
+          if hash != "" then
+            { outputHash = hash; }
+          else
+            {
+              outputHash = "";
+              outputHashAlgo = "sha256";
+            };
```

Comment:
What about just supporting hash?

Reply:
I am not sure what you mean by this. I thought FODs need an `outputHash` attr

Reply:
It is just supporting `hash`, otherwise this would be `if hash != "" && sha256 != ""`.